bezier-rs/lib/src/lib.rs

```diff
@@ -631,6 +631,41 @@ impl Bezier {
 			.collect::<Vec<DVec2>>()
 	}
 
+	/// Returns a list of lists of points representing the hull points for all iterations at the point corresponding to `t` using de Casteljau's algorithm
```

Comment:
"representing the hull points" -> what is a hull? What are hull points? What does this function actually do? This nicely describes how to interpret the data it returns, but doesn't explain what the algorithm is even for.

Reply:
One main reason why we're implementing this function is that it is in bezierjs.

Another is that `hull` can be used as an alternate implementation for `split` and `trim`. An intuitive visual representation of how that would work is available [here](https://pomax.github.io/bezierinfo/#splitting).

To be honest though, I'm not sure whether using `hull` is more or less performant than our current more algebraic approach (which is described [here](https://pomax.github.io/bezierinfo/#matrixsplit)). I think it boils down to whether a recursive solution is better than something involving more complex computations (which is also possible to rewrite to matrix multiplications). @TrueDoctor , do you have any input on this topic?